assembl/models/langstrings.py

```diff
@@ -464,6 +464,49 @@ def EMPTY(cls, db=None):
     def reset_cache(cls):
         pass
 
+    # Which object owns this?
+    owner_object = None
+    _owning_relns = []
+
+    @classmethod
+    def setup_ownership_load_event(cls, owner_class, relns):
+        def load_owner_object(target, context):
+            for reln in relns:
+                if reln in target.__dict__:
+                    getattr(target, reln).owner_object = target
+        event.listen(owner_class, "load", load_owner_object, propagate=True)
+        event.listens_for(owner_class, "refresh", load_owner_object, propagate=True)
+        def set_owner_object(target, value, old_value, initiator):
+            if old_value is not None:
+                old_value.owner_object = None
+            value.owner_object = target
+        for reln in relns:
+            cls._owning_relns.append((owner_class, reln))
+            event.listen(getattr(owner_class, reln), "set", set_owner_object, propagate=True)
+
+    def get_owner_object(self):
+        if self.owner_object is None and inspect(self).persistent:
+            self.owner_object = self.owner_object_from_query()
+        return self.owner_object
+
+    def owner_object_from_query(self):
+        queries = []
+        for owning_class, reln_name in self._owning_relns:
+            backref_name = owning_class.__mapper__.relationships[reln_name].backref[0]
```

Comment:
The sqla backref structure is an unnamed tuple, not much I can do about the [0].
But yes I assume there's a backref.
That would fail early at worst.